### DISCUSSION TITLE
Valid creation of IPAM / IP address relations.

### DISCUSSION THREAD

**Discussion by xcdr** [netbox-community/netbox#6074] [General]
Hi.

I use pynetbox for some glue between PROXMOX and NEBOX.

Part of my code for creating address instance looks that:
```
 addr = nb.ipam.ip_addresses.create(
        address=iface_addr,
        tenant=cluster.tenant.id,
        assigned_object_type = 'virtualization.vminterface',
        assigned_object_id = iface.id
 )
```

This is consistent with JSON presented by API.

But when I want search for address, I must search with similar filter:
```
current_addresses = nb.ipam.ip_addresses.filter(
        vminterface_id = iface.id
)
```
Discussion about this behavior is in issue: https://github.com/netbox-community/netbox/issues/5997

I cannot find documentation about it and my question is - How I should correctly creating address instances and why filter is inconsistent with keys presented by API.

**Comment by candlerb** [comment]
See #6489.

> How I should correctly creating address instances

Looks like you're doing it correctly already.

> why filter is inconsistent with keys presented by API

The REST API query parameters are documented via the online documentation at `/api/docs/`.

As far as I can see, they're not consistent with the attribute names because each individual query parameter generates its own Django filter in the code [here](https://github.com/netbox-community/netbox/blob/v3.0.2/netbox/ipam/filtersets.py#L446-L541).  In other words, it doesn't seem possible to have a pair of parameters working together as if they were a single parameter.
